snapshot-labs/snapshot
# fix: fix missing tokens after shortening invalid domains

### Summary

1. When shortening invalid ENS domains on space creation page, it does not take into account subdomains, so `0x1234.b.c` will be shortened into something like `[0x...34].b`
2. Some invalid domains may not have the invalid string `[xxxx]`, so shortening will not be necessary.

Fixes:

1. Taking into account subdomain, so `0x1234.b.c` will return `[0x1234].b.c`
2. Do not shorten invalid ENS domain when not containing `[` and `]`

### How to test

Somewhat difficult to test, as I don't have invalid domains ready.

Best way will be to edit the useEns composable to return a static list of invalid domains, then see the results in the space creation page

Example, inject following code in useENS.ts, at line 14

```
ownedEnsDomains.value = [
      { name: 'test.eth', isInvalid: true },
      { name: '[testfsdsdsdf].test.eth', isInvalid: true },
      { name: 'test.test.eth', isInvalid: true }
    ];

    return true;
```
